spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -1017,4 +1019,38 @@ public String unknown(
       return String.format("%s(%s) %s %s", transform, sourceName, direction, nullOrder);
     }
   }
+
+  public static Map<String, String> buildWriteOptions(String branch, LogicalWriteInfo info) {
```

Comment:
I don't think that we can safely support a `branch` write option, so I'm not sure why we would pass the branch as a write option internally. Why not pass the branch into `SparkWriteConf` directly? Then we don't need this method and don't need to worry about future updates that allow a branch option from Spark to be used.